db/controllers/controllers.js

```diff
@@ -46,8 +47,13 @@ exports.getCommentsByArticleId = (req, res, next)=>{
     })
 }
 exports.getAllArticles = (req,res,next)=>{
-    fetchAllArticles()
-    .then((articles)=>{
+   const {topic}=req.query
+
+    const promises=[fetchAllArticles(topic)]
+    if(topic){ promises.push(fetchTopicsBySlug(topic))}
```

Comment:
At this stage we should only need one model to be queried, what's the reason why we are querying two models here? With the tests that you have written we could rely on the one model and check the response for our error handling. Have a think about what tests we have written so far and what code those tests are driving us to write.